src/internal_events/wasm/compilation.rs

```diff
@@ -5,76 +5,80 @@ use metrics::counter;
 use std::time::{Duration, Instant};
 use vector_wasm::Role;
 
-#[derive(Debug, Clone, Copy, PartialEq, Eq)]
+#[derive(Debug, Clone, PartialEq, Eq)]
 #[must_use]
-pub struct WasmCompilation {
+pub struct WasmCompilationProgress {
     role: Role,
     state: State,
-    #[cfg(feature = "wasm-timings")]
+    error: Option<String>,
     epoch: Instant,
-    #[cfg(feature = "wasm-timings")]
     elapsed: Duration,
 }
 
-impl WasmCompilation {
+impl WasmCompilationProgress {
     pub fn begin(role: Role) -> Self {
         let me = Self {
             state: State::Beginning,
             role,
-            #[cfg(feature = "wasm-timings")]
+            error: Default::default(),
             epoch: Instant::now(),
-            #[cfg(feature = "wasm-timings")]
             elapsed: Default::default(),
         };
-        emit!(me);
+        emit!(me.clone());
         me
     }
     pub fn complete(self) {
         emit!(Self {
             state: State::Completed,
             role: self.role,
-            #[cfg(feature = "wasm-timings")]
+            error: self.error,
+            epoch: self.epoch,
```

Comment:
Could these be reduced down to `..self`?

Reply:
Way nicer. :)